include/realizations/catchment/Bmi_Module_Formulation.hpp

```diff
@@ -0,0 +1,937 @@
+         *
+         * Function uses AORC forcings from the "current" time step.
+         *
+         * @return Calculated ET or potential ET, or ``0.0`` if required parameters could not be obtained.
+         */
+        double calc_et() override {
+            return calc_et(
+                    get_ts_index_for_time(convert_model_time(get_bmi_model()->GetCurrentTime()) +
+                                          get_bmi_model_start_time_forcing_offset_s()));
+        }
+
+        double calc_et(struct AORC_data raw_aorc) {
+            // TODO: fix both not checking this, and how the way it's set currently doesn't fit with these kernels
+            //if (!is_et_params_set()) {
+            //    throw std::runtime_error("Can't calculate ET for BMI model without ET params being set");
+            //}
+            // TODO: the Et_Accountable and Et_Aware interfaces need to be overhauled, and these things need to all be
+            //  moved under whatever type is used for "et_parameters"
+
+            /*
+             * ********************************************************************************************************
+             * Logic based largely on et_wrapper_function() in EtWrapperFunction.hpp, and a little from et_setup() in
+             * EtSetParams.hpp.
+             * ********************************************************************************************************
+             */
+
+            // TODO: Putting made-up values here, but need to figure out where to actually get these from (these taken
+            //  from EtSetParams.hpp, with comments from relevant lines quoted)
+            double made_up_canopy_resistance_sec_per_m = 50.0; // "from plant growth model"
+            double made_up_water_temperature_C = 15.5; // "from soil or lake thermal model"
+            double made_up_ground_heat_flux_W_per_sq_m = -10.0; /* 0.0; */ // -40 "from soil thermal model. Negative denotes downward."
+            double made_up_vegetation_height_m = 0.12; // "used for unit test of aerodynamic resistance used in Penman Monteith method."
+            double made_up_zero_plane_displacement_height_m = 0.0003; /* 0.1; */  // "0.03 cm for unit testing"
+            double made_up_surface_longwave_emissivity = 1.0; // "this is 1.0 for granular surfaces, maybe 0.97 for water"
+            double made_up_surface_shortwave_albedo = 0.22; /* 0.0; */ // "this is a function of solar elev. angle for most surfaces."
+            double made_up_surface_skin_temperature_C = 12.0; /* 20.0; */  // "from soil thermal model or vegetation model"
+            double made_up_momentum_transfer_roughness_length_m = 0.0; /* 1.0; */ // zero means that default values will be used in routine.
+            double made_up_heat_transfer_roughness_length_m = 0.0; /* 1.0; */ // zero means that default values will be used in routine.
+
+            // TODO: for now, hard-code these things
+            struct et::evapotranspiration_options et_options;
```

Comment:
We should add some initializer functions to the et namespace and move this "model specific" code there and simply pass in an empty struct reference and let them be set there.

Reply:
This is now tracked in #276.